### PR TITLE
ci: unfreeze for master

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

It is fine to freeze for PR to have control env, but to prevent some issues and unexpected dependency breakages, let's run tests after the merge to master without freeze.

it shall bring visibility to cases like #17106

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda @carmocca @akihironitta @tchaton @justusschock @awaelchli

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

File: src/lightning_app/__setup__.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 _SOURCE_ROOT = os.path.join(_PROJECT_ROOT, "src")
 _PACKAGE_ROOT = os.path.join(_SOURCE_ROOT, "lightning_app")
 _PATH_REQUIREMENTS = os.path.join("requirements", "app")
-_FREEZE_REQUIREMENTS = bool(int(os.environ.get("FREEZE_REQUIREMENTS", 0)))
+_FREEZE_REQUIREMENTS = bool(os.environ.get("FREEZE_REQUIREMENTS", 0))
```
Comment (carmocca) [inline src/lightning_app/__setup__.py:14]: These changes don't look right: `bool("0") is True`

File: .github/workflows/ci-tests-pytorch.yml [hunk 2]
```diff
@@ -22,15 +22,14 @@ on:
       - "!requirements/pytorch/docs.txt"
       - "!*.md"
       - "!**/*.md"
+  schedule:
```
Comment (carmocca) [inline .github/workflows/ci-tests-pytorch.yml:25]: What are the implications of this change?

**Review by carmocca — APPROVED** [review]

File: src/lightning_app/__setup__.py (hunk 1 above)
Comment (Borda) [reply src/lightning_app/__setup__.py:14]: dam, i missed it, thx for catching it :)

File: .github/workflows/ci-tests-pytorch.yml (hunk 2 above)
Comment (Borda) [reply .github/workflows/ci-tests-pytorch.yml:25]: In case we don't merge anything for a day, we still get daily glances what is the actual state

**Review by akihironitta — APPROVED** [review]

**Pull request merged**